src/checkers/inference/model/ConstraintManager.java

```diff
@@ -55,109 +50,91 @@ public ConstraintVerifier getConstraintVerifier() {
         return constraintVerifier;
     }
 
+    /**
+     * Adds the given constraint to the constraint set if the ignoreConstraints flag is set to
+     * false, and the constraint is not an {@link AlwaysTrueConstraint}. An error is issued if the
+     * constraint is an {@link AlwaysFalseConstraint}.
+     */
     private void add(Constraint constraint) {
         if (!ignoreConstraints) {
-            constraints.add(constraint);
+            if (constraint instanceof AlwaysFalseConstraint) {
+                ErrorReporter.errorAbort(
+                        "Attempting to add an AlwaysFalseConstraint to the constraint set.");
+            } else if (!(constraint instanceof AlwaysTrueConstraint)) {
+                constraints.add(constraint);
+            }
         }
     }
 
     public void startIgnoringConstraints() {
-        this.ignoreConstraints = true;
+        ignoreConstraints = true;
     }
 
     public void stopIgnoringConstraints() {
-        this.ignoreConstraints = false;
+        ignoreConstraints = false;
     }
 
-    public SubtypeConstraint createSubtypeConstraint(Slot subtype, Slot supertype) {
-        if (subtype == null || supertype == null) {
-            ErrorReporter.errorAbort("Create subtype constraint with null argument. Subtype: " + subtype
-                    + " Supertype: " + supertype);
-        }
-        if (subtype instanceof ConstantSlot && supertype instanceof ConstantSlot) {
-            ConstantSlot subConstant = (ConstantSlot) subtype;
-            ConstantSlot superConstant = (ConstantSlot) supertype;
+    // All createXXXConstraint methods create a (possibly normalized) constraint for the given
+    // slots, checking for null arguments. It does not issue errors for unsatisfiable constraints.
```

Comment:
"checking for null arguments" : in your implementation, it seems `create` methods in Constraint checking this, not `createXXXConstraint` methods. Maybe remove this code comment for `createXXXConstraint` ?

Reply:
Good catch. It was before I migrated the null arg checks. Will fix.